synapse/state/__init__.py

```diff
@@ -554,16 +555,20 @@ async def resolve_state_groups(
             if conflicted_state:
                 logger.info("Resolving conflicted state for %r", room_id)
                 with Measure(self.clock, "state._resolve_events"):
-                    # mypy doesn't like that new_state was previously mutable,
-                    # and now isn't. We don't mutate past this point though, so
-                    # tell it to go away.
-                    new_state = await resolve_events_with_store(  # type: ignore
-                        self.clock,
-                        room_id,
-                        room_version,
-                        list(state_groups_ids.values()),
-                        event_map=event_map,
-                        state_res_store=state_res_store,
+                    # resolve_eevnts_with_store returns a StateMap, but we can
```

Comment:
```suggestion
                    # resolve_events_with_store returns a StateMap, but we can
```

I like the new comment, it gives me more confidence.